### PR TITLE
Indexing test for starting a 3.2 db on 3.3

### PR DESCRIPTION
**Pull request by tinwelint**
It also creates new indexes so that there are Lucene and Fusion indexes
living side by side.

### CODE REVIEW

File: community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        GraphDatabaseService db = new GraphDatabaseFactory().newEmbeddedDatabase( storeDir );
+        try
+        {
+            createIndexesAndData( db, LABEL1 );
+        }
+        finally
+        {
+            db.shutdown();
+        }
+        System.out.println( "Db created in " + storeDir.getAbsolutePath() );
```
Comment (burqen) [inline community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java:86]: System.out 🎉  (although in ignored test, but anyway)

File: community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            }
+            tx.success();
+        }
+    }
+
+    private void verifyIndexes( GraphDatabaseAPI db, Label label ) throws Exception
+    {
+        // There should be an index for the label and KEY1 containing 50 nodes
+        assertTrue( hasIndex( db, label, KEY1 ) );
+        assertEquals( 100, countIndexedNodes( db, label, KEY1 ) );
```
Comment (burqen) [inline community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java]: Comment says 50, code says 100 nodes.

File: community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+        // There should be an index for the label and KEY1+KEY2 containing 33 nodes
+        assertTrue( hasIndex( db, label, KEY1, KEY2 ) );
+        assertEquals( 34, countIndexedNodes( db, label, KEY1, KEY2 ) );
```
Comment (burqen) [inline community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java]: Comment says 33, code says 34 nodes.

**Review by burqen — APPROVED** [review]
Sweet test. Just got some small comments.

File: community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java (hunk 1 above)
Comment (tinwelint) [reply community/neo4j/src/test/java/migration/Start3_2DbOn3_3AndCreateFusionIndexIT.java:86]: It's intended... you want that when creating the db :) something you basically never do, but there as a reference for how the db was created for 3.2

**Pull request merged**